staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator.go

```diff
@@ -192,6 +204,7 @@ func (a *cachedTokenAuthenticator) doAuthenticateToken(ctx context.Context, toke
 
 		record.resp, record.ok, record.err = a.authenticator.AuthenticateToken(ctx, token)
 		record.annotations = ev.Annotations
+		record.warnings = recorder.warnings
```

Comment:
just to be safe, move `warnings` from `recorder` to `record` under lock and null out `recorder.warnings` field so any late-arriving async calls to AddWarning don't modify the shared `warnings` var

something like:

```go
func (r *recorder) extractWarnings() []*cacheRecord {
	r.mu.Lock()
	defer r.mu.Unlock()
  warnings := r.warnings
  r.warnings = nil
  return warnings
}
```

used like 

```go
record.warnings = recorder.extractWarnings()
```

Reply:
done